src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/custom.py

```diff
@@ -77,58 +76,26 @@ def show_storage_account_connection_string(
     """ Generate connection string for a storage account."""
     from azure.cli.core._profile import CLOUD
     scf = storage_client_factory()
-    keys = scf.storage_accounts.list_keys(resource_group_name, account_name).keys  # pylint: disable=no-member
+    obj = scf.storage_accounts.list_keys(resource_group_name, account_name)  # pylint: disable=no-member
+    try:
+        keys = [obj.keys[0].value, obj.keys[1].value]  # pylint: disable=no-member
+    except AttributeError:
+        # Older API versions have a slightly different structure
+        keys = [obj.key1, obj.key2]  # pylint: disable=no-member
```

Comment:
This is the only line that's different and changes like this will not happen often.
Not sure if duplicating the whole method for this one change is the right way to go.